inc/notificationtemplate.class.php

```diff
@@ -430,20 +434,24 @@ static function processIf($string, $data) {
 
                   $condition_ok = true;
 
-               } else {
-                  $condition_ok = false;
                }
-
             } else { // check exact match
+               if (isset($data['##'.$if_field.'##'])) {
+                  // Data value: the value for the field in the database
+                  $data_value = Html::entity_decode_deep($data['##'.$if_field.'##']);
+
+                  // Condition value: the expected value needed to validate the condition
+                  $condition_value = Html::entity_decode_deep($out[2][$key]);
+
+                  // Special case for data returned by Dropdown::getYesNo, we
+                  // need to use the localized value in the comparison
+                  $to_translate = ["Yes", "No"];
+                  if (in_array($condition_value, $to_translate)) {
```

Comment:
Problem may also occured on other fields (e.g. `$tmp['##task.status##']       = Planning::getState($task['state']);`).
So why not try to translate every value ?

By the way, I do not know if checking condition only on translated value is a good point. Maybe check should be done on both based and translated value.

Reply:
> By the way, I do not know if checking condition only on translated value is a good point. Maybe check should be done on both based and translated value.

Correct me if I'm wrong but with the current notification model I'm not sure we have access to the raw value ?
Obviously the best way would be to compare directly on these raw values but without reworking our current model it does not seem possible.

Reply:
I was talking about comparing value with the translated one and the not translated one, i.e. the key 'Yes' and the translated 'Oui' in french.

Reply:
By the way, is this `Yes` value an untranslated value or a value translated in the default language defined in GLPI installation process ? If it comes from the default language, your fix may only work in specific installations.

Reply:
The values are translated to the current language:

```php
$tmp['##followup.isprivate##']   = Dropdown::getYesNo($followup['is_private']);
```


```php
static function getYesNo($value) {
  if ($value) {
     return __('Yes');
   }
   return __('No');
}
```